contentcuration/contentcuration/frontend/shared/views/channel/RemoveChannelModal.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+
+  <KModal
+    v-if="dialog"
+    :title="canEdit ? $tr('deleteTitle') : $tr('removeTitle')"
+    :submitText="canEdit ? $tr('deleteChannel') : $tr('removeBtn')"
+    :cancelText="$tr('cancel')"
+    :submitDisabled="loading"
+    data-test="remove-channel-modal"
+    @submit="handleSubmit"
+    @cancel="close"
+  >
+    <div
+      v-if="loading"
+      class="py-4 text-center"
+    >
+      <VProgressCircular
+        indeterminate
+        size="24"
+      />
+    </div>
+    <template v-else>
+      <div
+        v-if="canEdit && hasCommunityLibrarySubmission"
+        class="mb-3"
+        data-test="cl-warning"
+      >
+        {{ $tr('deleteChannelWithCLWarning') }}
+      </div>
+      {{ canEdit ? $tr('deletePrompt') : $tr('removePrompt') }}
+    </template>
+  </KModal>
+
+</template>
+
+
+<script>
+
+  import { defineComponent, ref, computed, watch, getCurrentInstance } from 'vue';
+  import client from 'shared/client';
+
+  export default defineComponent({
```

Comment:
Oh, we don't need here the `defineComponent` function, since we are working with component files, Vue automatically interprets that the default export of this file is the definition of the component.